iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -454,6 +416,190 @@ public async Task RetrySetRetryPolicyVerifyInternalsSuccess()
+                .Setup(x => x.OpenAsync(It.IsAny<CancellationToken>()))
+                .Returns(() => Task.CompletedTask);
+
+            using var retryDelegatingHandler = new RetryDelegatingHandler(contextMock, nextHandlerMock.Object);
+
+            await retryDelegatingHandler.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+            await retryDelegatingHandler.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+
+            // act
+            Func<Task> open = () => retryDelegatingHandler.OpenAsync(CancellationToken.None);
+
+            // assert
+            await open.Should().NotThrowAsync().ConfigureAwait(false); ;
+        }
+
+        [TestMethod]
+        public async Task RetryDelegatingHandler_SendTelemetryAsyncAfterCloseAsync_Throws()
+        {
+            // arrange
+            var contextMock = new PipelineContext
+            {
+                RetryPolicy = new TestRetryOncePolicy(),
+                ConnectionStatusChangeHandler = (connectionStatusInfo) => { }
+            };
+
+            var nextHandlerMock = new Mock<IDelegatingHandler>();
+            nextHandlerMock
+                .Setup(x => x.OpenAsync(It.IsAny<CancellationToken>()))
+                .Returns(() => Task.CompletedTask);
+
+            using var retryDelegatingHandler = new RetryDelegatingHandler(contextMock, nextHandlerMock.Object);
+
+            await retryDelegatingHandler.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+            await retryDelegatingHandler.CloseAsync(CancellationToken.None).ConfigureAwait(false);
+
+            // act
+            Func<Task> sendTelemetry = () => retryDelegatingHandler.SendTelemetryAsync(It.IsAny<TelemetryMessage>(), CancellationToken.None);
+
+            // assert
+            await sendTelemetry.Should().ThrowAsync<InvalidOperationException>().ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task RetryDelegatingHandler_OpenAsyncAfterDispose_Throws()
+        {
+            // arrange
+            var contextMock = new PipelineContext
+            {
+                RetryPolicy = new TestRetryOncePolicy(),
+                ConnectionStatusChangeHandler = (connectionStatusInfo) => { }
+            };
+
+            var nextHandlerMock = new Mock<IDelegatingHandler>();
+            nextHandlerMock
+                .Setup(x => x.OpenAsync(It.IsAny<CancellationToken>()))
+                .Returns(() => Task.CompletedTask);
+
+            using var retryDelegatingHandler = new RetryDelegatingHandler(contextMock, nextHandlerMock.Object);
+
+            await retryDelegatingHandler.OpenAsync(CancellationToken.None).ConfigureAwait(false);
+            retryDelegatingHandler.Dispose();
+
+            // act
+            Func<Task> open = () => retryDelegatingHandler.OpenAsync(CancellationToken.None);
+
+            // assert
+            await open.Should().ThrowAsync<ObjectDisposedException>().ConfigureAwait(false);
+        }
+
+        [Ignore]
```

Comment:
This test currently throws as NRE indicating the semaphores have been disposed and dereferenced. With the cooperative cancellation update on CloseAsync(), this functionality will be modified and the SDK should then expect an ObjectDisposedException instead. We will enable the test once the update has been added in.